core/templates/cowdata.h

```diff
@@ -65,24 +65,26 @@ class CowData {
 	friend class VMap;
 
 private:
+	static constexpr uint64_t ALLOC_PAD = sizeof(uint64_t) * 2;
+
 	mutable T *_ptr = nullptr;
 
 	// internal helpers
 
-	_FORCE_INLINE_ SafeNumeric<uint32_t> *_get_refcount() const {
+	_FORCE_INLINE_ SafeNumeric<uint64_t> *_get_refcount() const {
 		if (!_ptr) {
 			return nullptr;
 		}
 
-		return reinterpret_cast<SafeNumeric<uint32_t> *>(_ptr) - 2;
+		return reinterpret_cast<SafeNumeric<uint64_t> *>(_ptr) - 2;
 	}
 
-	_FORCE_INLINE_ uint32_t *_get_size() const {
+	_FORCE_INLINE_ uint64_t *_get_size() const {
 		if (!_ptr) {
 			return nullptr;
 		}
 
-		return reinterpret_cast<uint32_t *>(_ptr) - 1;
+		return reinterpret_cast<uint64_t *>(_ptr) - 1;
 	}
 
 	_FORCE_INLINE_ size_t _get_alloc_size(size_t p_elements) const {
```

Comment:
I'd suggest switching to `uint64_t` here as `size_t` isn't guaranteed to be 64 bits though it usually is, better stay safe IMO

Reply:
I ended up using a typedef for the size, as suggested.